<?xml version="1.0"?>
<launch>
  <arg name="joy_dev" default="/dev/input/js0" />
  <arg name="server" default="192.168.0.3" />
  <arg name="port" default="3883" />

  <arg name="uri1" default="radio://0/80/2M" />
  <arg name="frame1" default="crazyflie1" />

  <arg name="uri2" default="radio://0/60/2M" />
  <arg name="frame2" default="crazyflie2" />

  <arg name="uri3" default="radio://0/100/2M" />
  <arg name="frame3" default="crazyflie3" />-->

  <include file="$(find crazyflie_driver)/launch/crazyflie_server.launch">
  </include>

  <node name="joy" pkg="joy" type="joy_node" output="screen">
    <param name="dev" value="$(arg joy_dev)" />
  </node>

  <group ns="crazyflie1">
    <include file="$(find crazyflie_driver)/launch/crazyflie_add.launch">
      <arg name="uri" value="$(arg uri1)" />
      <arg name="tf_prefix" value="crazyflie1" />
      <arg name="enable_logging" value="False" />
      <arg name="pitch_trim" value="-0.0" />
      <arg name="roll_trim" value="0.0" />
    </include>

    <node name="crazyflie_demo_controller" pkg="crazyflie_demo" type="controller.py" output="screen">
      <param name="use_crazyflie_controller" value="True" />
      <param name="joy_topic" value="/joy" />
    </node>

    <include file="$(find crazyflie_controller)/launch/crazyflie_c1.launch">
      <arg name="frame" value="$(arg frame1)" />
    </include>

    <node name="publish_pose" pkg="crazyflie_demo" type="goaldynamic_advanced.py" output="screen">
      <param name="name" value="goal" />
      <param name="rate" value="100.0" />
      <param name="myframe" value="$(arg frame1)" />

    </node>
  </group>

  <group ns="crazyflie2">
    <include file="$(find crazyflie_driver)/launch/crazyflie_add.launch">
      <arg name="uri" value="$(arg uri2)" />
      <arg name="tf_prefix" value="crazyflie2" />
      <arg name="enable_logging" value="False" />
      <arg name="pitch_trim" value="-0.25" />
      <arg name="roll_trim" value="-0.1" />
    </include>

    <node name="crazyflie_demo_controller" pkg="crazyflie_demo" type="controller.py" output="screen">
      <param name="use_crazyflie_controller" value="True" />
      <param name="joy_topic" value="/joy" />
    </node>

    <include file="$(find crazyflie_controller)/launch/crazyflie_c2.launch">
      <arg name="frame" value="$(arg frame2)" />
    </include>

    <node name="publish_pose" pkg="crazyflie_demo" type="goaldynamic_advanced.py" output="screen">
      <param name="name" value="goal" />
      <param name="rate" value="100.0" />
      <param name="myframe" value="$(arg frame2)" />

    </node>
  </group>

  <group ns="crazyflie3">
    <include file="$(find crazyflie_driver)/launch/crazyflie_add.launch">
      <arg name="uri" value="$(arg uri3)" />
      <arg name="tf_prefix" value="crazyflie3" />
      <arg name="enable_logging" value="False" />
      <arg name="pitch_trim" value="-0.0" />
      <arg name="roll_trim" value="-0.0" />
    </include>

    <node name="crazyflie_demo_controller" pkg="crazyflie_demo" type="controller.py" output="screen">
      <param name="use_crazyflie_controller" value="True" />
      <param name="joy_topic" value="/joy" />
    </node>

    <include file="$(find crazyflie_controller)/launch/crazyflie_c3.launch">
      <arg name="frame" value="$(arg frame3)" />
    </include>

    <node name="publish_pose" pkg="crazyflie_demo" type="goaldynamic_advanced.py" output="screen">
      <param name="name" value="goal" />
      <param name="rate" value="100.0" />
      <param name="myframe" value="$(arg frame3)" />

    </node>
  </group>

  <!-- run vrpn client -->
  <node pkg="vrpn_client_ros" type="vrpn_client_node" name="vrpn_client_node" output="screen">
    <rosparam subst_value="true">
    server: $(arg server)
    port: $(arg port)
    update_frequency: 100.0
    frame_id: world
    use_server_time: false
    broadcast_tf: true
    refresh_tracker_frequency: 1.0 <!-- VRPN will add all trackers found on connection -->
    </rosparam>
   </node>


</launch>
